subsys/net/ip/net_context.c

```diff
@@ -689,6 +689,45 @@ int net_context_put(struct net_context *context)
 	return 0;
 }
 
+static int net_context_bind_default(struct net_context *context,
+				    const struct sockaddr *dest_addr)
+{
+	struct net_if *iface = net_if_get_default();
+
+	if (!iface) {
+		NET_ERR("Cannot bind to %s",
+			net_sprint_ipv6_addr(&addr6->sin6_addr));
+
+		return -EADDRNOTAVAIL;
+	}
+
+	if (dest_addr->family == AF_INET6) {
+		if (net_sin6_ptr(&context->local)->sin6_addr) {
+			return 0;
+		}
+
+		net_context_set_iface(context, iface);
+		net_sin6_ptr(&context->local)->sin6_family = AF_INET6;
+		net_sin6_ptr(&context->local)->sin6_addr =
+			(struct in6_addr *)net_ipv6_unspecified_address();
+
```

Comment:
Let's treat this as a separate issue. So far, I indeed rely that random port was set at the time of context creation. As you say, that's actually too early, and may be already used at the time of actual network operation. But the obvious solution is to *move* setting random port to bind_default(), which is a separate patch, requiring separate consideration and testing (UDP cases, etc.)